### PR TITLE
Add a logical failure mask to HwError::I2cError

### PR DESCRIPTION
**Pull request by Aaron-Hartwig**
This is part of the efforts to address https://github.com/oxidecomputer/hubris/issues/1069

### CODE REVIEW

File: messages/src/lib.rs [hunk 1]
```diff
@@ -267,12 +267,13 @@ pub enum HwError {
...
-    /// The FPGA reported an I2C error
+    /// The FPGA reported an I2C error.  u32 is a logical mask for which
+    /// ports saw a failure.
     #[cfg_attr(
         any(test, feature = "std"),
-        error("FPGA reported an I2C error, module may not be present")
+        error("FPGA reported an I2C error, module may not be present. Logical mask for which ports saw a failure: 0x{0:08x}")
     )]
-    I2cError,
+    I2cError(u32),
```
[the remark this answers is not in the document]
Comment (bnaecker) [reply messages/src/lib.rs:276]: Actually, it seems like this should just be `PortMask`, no?

File: messages/src/lib.rs [hunk 2]
```diff
@@ -267,12 +267,13 @@ pub enum HwError {
     )]
     WaitFailed,
 
-    /// The FPGA reported an I2C error
+    /// The FPGA reported an I2C error.  u32 is a logical mask for which
+    /// ports saw a failure.
     #[cfg_attr(
         any(test, feature = "std"),
-        error("FPGA reported an I2C error, module may not be present")
+        error("FPGA reported an I2C error, module may not be present. Logical mask for which ports saw a failure: 0x{0:08x}")
```
Comment (bnaecker) [inline messages/src/lib.rs:274]: I would personally love to see this error report the actual indices of the modules which reported failure. The value in the `I2cError` enum variant can and should still be the bit-mask type, but the human-level error that we print, for example in `xcvradm` could be just the list of indices.

[Here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=00fa5467289f99e35e70d51d5d86e711) is a playground link that shows how you'd do this. `thiserror` is extremely flexible, and we can use the `PortMask::to_indices` method to turn the bitmask into a list of indices in the error message printed. Hopefully that makes sense, and let me know if that seems reasonable!

File: messages/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aaron-Hartwig) [reply messages/src/lib.rs:276]: Yeah following the conversation we had yesterday evening on logical vs physical mask I agreed that a logical mask isn't useful at this layer, so a `u16` is definitely appropriate. I will make this `PortMask`.

File: messages/src/lib.rs (hunk 2 above)
Comment (Aaron-Hartwig) [reply messages/src/lib.rs:274]: I'll dig into `thiserror` more and get it to print out each individual index. That seems like a very reasonable and helpful thing to do!

**Comment by Aaron-Hartwig** [comment]
@bnaecker with your latest feedback I get something like this now:
```
aaron@niles ~ $ ./xcvradm -i axf2 -f 0 -t 0-2 status
FPGA Port Status
   0    0 PRESENT | ENABLED | LOW_POWER_MODE | INTERRUPT | POWER_GOOD
   0    1 PRESENT | ENABLED | RESET | LOW_POWER_MODE | POWER_GOOD
   0    2 PRESENT | ENABLED | LOW_POWER_MODE | INTERRUPT | POWER_GOOD
aaron@niles ~ $ ./xcvradm -i axf2 -f 0 -t 0-2 identify
Error: Failed to identify transceiver modules

Caused by:
    0: Network protocol error
    1: Failure during read: FPGA reported an I2C error at the following indices: [1]
aaron@niles ~ $ ./xcvradm -i axf2 -f 0 identify
Error: Failed to identify transceiver modules

Caused by:
    0: Network protocol error
    1: Failure during read: FPGA reported an I2C error at the following indices: [1, 3, 5, 7, 9, 11, 13, 14, 15]
```

**Comment by bnaecker** [comment]
Tiny nit, but I might slightly reword the error message. Instead of `at the following indices`, maybe `on the following ports` would be a bit more clear. Other than that LGTM, nice work!

**Comment by Aaron-Hartwig** [comment]
@bnaecker I fixed that up in https://github.com/oxidecomputer/transceiver-control/pull/60/commits/55ad737f9dd062ca122bf715396272988b36942f so I'm going to merge this.

**Pull request merged**